eodhp_utils/pulsar/client.py

```diff
@@ -0,0 +1,52 @@
+import json
+import os
+
+from pulsar import Client as PulsarClient
+
+
+class Client:
+    def __init__(self, url=None):
+        if url is None:
+            url = os.environ.get("PULSAR_URL")
+        self.client = PulsarClient(url)
+
+
+class Producer(Client):
+    def __init__(self, topic, producer):
+        super().__init__()
+
+        self.producer = self.client.create_producer(topic=topic, producer_name=producer)
+
+    def send(self, message: bytes):
+        """Send a message. preferred format is bytes but also converts str and dict"""
```

Comment:
Your mesage arg is annotated with type bytes, but you are expecting it to either be a dict or a str. I would leave the conversion to the calling code instead of trying to handle all eventualities in your client.